[CI] Failure in DateFormattersTests testEqualsAndHashcode

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/123/console

Reproduces for me locally too with:

```
REPRODUCE WITH: ./gradlew :server:test \
  -Dtests.seed=CE83486EF607DCDA \
  -Dtests.class=org.elasticsearch.common.time.DateFormattersTests \
  -Dtests.method="testEqualsAndHashcode" \
  -Dtests.security.manager=true \
  -Dtests.locale=fr-FR \
  -Dtests.timezone=Singapore \
  -Dcompiler.java=11 \
  -Druntime.java=8
```

Failure:

```
Note: Recompile with -Xlint:unchecked for details.
Suite: org.elasticsearch.common.time.DateFormattersTests
  1> [2018-10-06T02:56:19,891][INFO ][o.e.c.t.DateFormattersTests] [testEqualsAndHashcode] before test
  1> [2018-10-06T02:56:20,016][INFO ][o.e.c.t.DateFormattersTests] [testEqualsAndHashcode] after test
  2> REPRODUCE WITH: ./gradlew :server:test -Dtests.seed=CE83486EF607DCDA -Dtests.class=org.elasticsearch.common.time.DateFormattersTests -Dtests.method="testEqualsAndHashcode" -Dtests.security.manager=true -Dtests.locale=fr-FR -Dtests.timezone=Singapore -Dcompiler.java=11 -Druntime.java=8
FAILURE 0.17s | DateFormattersTests.testEqualsAndHashcode <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: not <format[YYYY] locale[]>
   >      but: was <format[YYYY] locale[]>
   >    at __randomizedtesting.SeedInfo.seed([CE83486EF607DCDA:BF8C30A339E095F5]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.common.time.DateFormattersTests.testEqualsAndHashcode(DateFormattersTests.java:114)
   >    at java.lang.Thread.run(Thread.java:748)
  2> NOTE: leaving temporary files on disk at: /home/brownbear/src/elasticsearch/server/build/testrun/test/J0/temp/org.elasticsearch.common.time.DateFormattersTests_CE83486EF607DCDA-001
  2> NOTE: test params are: codec=Asserting(Lucene80): {}, docValues:{}, maxPointsInLeafNode=1950, maxMBSortInHeap=5.398132832690126, sim=Asserting(org.apache.lucene.search.similarities.AssertingSimilarity@14f65ba5), locale=fr-FR, timezone=Singapore
  2> NOTE: Linux 4.15.0-36-generic amd64/Oracle Corporation 1.8.0_181 (64-bit)/cpus=16,threads=1,free=403589416,total=514850816
  2> NOTE: All tests run in this JVM: [DateFormattersTests]
Completed [1/1] in 0.93s, 1 test, 1 failure <<< FAILURES!

```

--- a/server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java
+++ b/server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java
@@ -111,8 +111,9 @@ public class DateFormattersTests extends ESTestCase {
         assertThat(DateFormatters.forPattern("YYYY").withZone(ZoneId.of("CET")), not(equalTo(DateFormatters.forPattern("YYYY"))));
 
         // different locale, thus not equals
-        assertThat(DateFormatters.forPattern("YYYY").withLocale(randomLocale(random())),
-            not(equalTo(DateFormatters.forPattern("YYYY"))));
+        DateFormatter f1 = DateFormatters.forPattern("YYYY").withLocale(Locale.CANADA);
+        DateFormatter f2 = f1.withLocale(Locale.FRENCH);
+        assertThat(f1, not(equalTo(f2)));
 
         // different pattern, thus not equals
         assertThat(DateFormatters.forPattern("YYYY"), not(equalTo(DateFormatters.forPattern("YY"))));

